src/dmd/traits.d

```diff
@@ -1416,11 +1416,20 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         if (dim != 1)
             return dimError(1);
 
+        void usage()
+        {
+            e.errorSupplemental("In `__traits( %s, TraitsArgument )` `TraitsArgument` must evaluate to either a type or an expression of type", e.ident.toChars());
+            e.errorSupplemental("Possible variants are module, struct, union, class, interface and template (not template declaration)");
+            e.errorSupplemental("For more info see https://dlang.org/spec/traits.html#%s", e.ident.toChars());
```

Comment:
A link to the spec is overkill IMHO because the valid options are listed above and they just clutter the error output (as seen in `fail_compilation/traits.d`)

Reply:
I don't mind removing it.